### PR TITLE
[MXNet-744] Fix website build pipeline Python 3 issues

### PR DESCRIPTION
**Pull request by aaronmarkham**
## Description ##
Some Python 3 incompatibilities existed in the website build pipeline. When CI was updated to run docs build in Python 3 mode these surfaced. This PR addresses it and locks in the Python dependency versions to help prevent future breakage.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
* Fixed Sphinx plugin mxdoc.py Python 3 issues by using `six`.
* Added `six` to dependencies
* Consolidated Python requirements for docs to one file in `ci/docker/install`
* Pinned dependencies to specific version numbers
* `mock` was not pinned upstream from docs, so we were getting a later version than what docs had been pinned to, so this is upgraded.
* "Upgraded" a variety of packages that were not pinned to the latest in PyPI. Tests indicate things are good, and I'd rather pin things now to prevent future breakage.
* Added a separate `Jenkinsfile-dev` so I don't have to keep hacking the Jenkinsfile every time I need to test something from my fork. This way I can just point to it from the test Jenkins jobs directly, and no more code updates. 
* Updated CI's docker images to use the new Python requirements file

## Comments ##
* We can break out a separate requirements file if some split in deps appears between Python 2/3. For now the packages are the same.

### CODE REVIEW

**Comment by aaronmarkham** [comment]
@marcoabreu @larroy Please take a look.

File: docs/Jenkinsfile-dev [hunk 1]
```diff
@@ -0,0 +1,54 @@
+// -*- mode: groovy -*-
```
Comment (nswamy) [inline docs/Jenkinsfile-dev]: where is this file used? why is it called Jenkinsfile-dev?

**Comment by nswamy** [comment]
LGTM, but I am not sure where is the new Jenkinsfile-dev used. I will merge this to unblock CI and can continue to follow up.

**Comment by aaronmarkham** [comment]
Login to Jenkins and look at test-website-build. That's what points to the Jenkinsfile-dev version. Email notifications for failures are off and it doesn't run code on restricted nodes. Might be a better way to do it, but seemed best for the situation.

**Pull request merged**